### PR TITLE
Update and add some methods to Kit base

### PR DESCRIPTION
**Pull request by sergigracia**
- [x] UnsplashKit base (Source)
- [x] Tests

### CODE REVIEW

File: UnsplashKit/Classes/Adapters/UrlDataResponseAdapter.swift [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import Foundation
 import Core
 
-public class UrlDataResponseAdapter: Adapter<(data: NSData?, response: NSURLResponse?), (NSData, NSURLResponse?)> {
+public class UrlDataResponseAdapter: Adapter<(data: NSData?, response: NSURLResponse?), (UIImage, NSURLResponse?)> {
```
Comment (pepicrft) [inline UnsplashKit/Classes/Adapters/UrlDataResponseAdapter.swift]: Use macros instead to not couple this to `UIKit`

File: UnsplashKit/Classes/Clients/Source/UnsplashSourceClient+Calls.swift [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import Foundation
+import RxSwift
+import SwiftyJSON
+
+extension UnsplashSourceClient {
+
+    public func randomPhoto(fromCategory category: SourceCategory, size: (Int, Int)? = nil) -> Observable<UIImage> {
+        return self.request(self.requestFactory.category(category)).map { $0.0 }
```
Comment (pepicrft) [inline UnsplashKit/Classes/Clients/Source/UnsplashSourceClient+Calls.swift]: 👏

**Comment by pepicrft** [comment]
👏 nice job!

**Comment by pepicrft** [comment]
Nice job! 
@sergigracia Travis was not passing :trollface:

**Comment by sergigracia** [comment]
It was green but Travis hook failed :P haha
Thanks man!

**Pull request merged**